File: Algorithm_136_Single Number.cpp
#include <iostream>
#include <vector>
#include <set>
using namespace std;
int singleNumber(vector<int>& nums)
{
	int size=nums.size();
    set<int> s;
    set<int>::iterator it;
    for (int i=0;i<size;i++)
    {
       it=s.find(nums[i]);
        if (it!=s.end()) s.erase(nums[i]);
        else s.insert(nums[i]);
    }
    return *s.begin();
}

File: Algorithm_136_Single Number.java
// 思路：bit manipulation。其他数都出现2次，因此异或后得到0。把nums中的所有数做异或，结果就是只出现一次的那个数

class Solution {
    public int singleNumber(int[] nums) {
        int res = nums[0];
        for (int i = 1; i < nums.length; i++) {
            res ^= nums[i];
        }
        return res;
    }
}